lib/Async/Registry/thread_registry.h

```diff
@@ -23,7 +23,9 @@ namespace arangodb::async_registry {
    This registry destroys itself when its ref counter is decremented to 0.
  */
 struct ThreadRegistry : std::enable_shared_from_this<ThreadRegistry> {
-  ThreadRegistry() = default;
+  static auto make() -> std::shared_ptr<ThreadRegistry> {
+    return std::shared_ptr<ThreadRegistry>(new ThreadRegistry{});
```

Comment:
Hm, I don't think that this is the same, because when I run this with ubsan, I get
```
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/shared_ptr_base.h:1351:9: runtime error: reference binding to null pointer of type 'element_type' (aka 'arangodb::async_registry::ThreadRegistry')
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/shared_ptr_base.h:1351:9 in 
/home/jvolmer/code/arangodb/lib/Async/Registry/registry_variable.cpp:17:10: runtime error: reference binding to null pointer of type 'element_type' (aka 'arangodb::async_registry::ThreadRegistry')
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/jvolmer/code/arangodb/lib/Async/Registry/registry_variable.cpp:17:10 in 
/home/jvolmer/code/arangodb/lib/Async/include/Async/async.h:54:43: runtime error: member call on null pointer of type 'arangodb::async_registry::ThreadRegistry'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/jvolmer/code/arangodb/lib/Async/include/Async/async.h:54:43 in 
/home/jvolmer/code/arangodb/lib/Async/Registry/thread_registry.h:36:8: runtime error: member call on null pointer of type 'arangodb::async_registry::ThreadRegistry *'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/jvolmer/code/arangodb/lib/Async/Registry/thread_registry.h:36:8 in 
AddressSanitizer:DEADLYSIGNAL
```
Possibly because the ThreadRegistry constructor is private?
I removed the commit again.

Reply:
Wow sorry for this, I think default ctor for shared_ptr is empty, ie. it is equivalent to `std::shared_ptr<ThreadRegistry>{nullptr}`(https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr) and that is not what you want